src/app/Directory.hpp

```diff
@@ -48,6 +53,21 @@ class Directory {
   Directory(const QString& path);
 
   /**
+   * Copy Constructor
+   */
+  Directory(const Directory& other);
+
+  /**
+   * Move Constructor.
+   */
+  Directory(Directory&& other) noexcept;
+
+  /**
+   * Copy assignment.
```

Comment:
This handles both move and copy. Should probably just say assignment and note this fact.